### PR TITLE
Added opc-input-chip component.

### PR DESCRIPTION
**Pull request by gisumit**
# Added opc-input-chip component.

## Screenshots

Screenshot 2020-12-01 at 3 14 29 PM

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] My changes generate no new warnings
- [x] I have performed a self-review of my own code
- [x] I have documented my code, particularly in areas like todo, complex logic, quick fix, temporary patch, etc.
#### If it is a new component
- [x] Does your component follow One Platform style guidelines?
- [x] Does your component web accessibility standards? [Helper Doc](https://www.w3.org/TR/wai-aria-1.1/)
- [x] Does your component support desktop screen sizes (350px, 720px, 1150px ,1920px)
#### Browsers you have tested in
- [x] Latest two versions of Mozilla Firefox and Google Chrome supported by Red Hat Enterprise Linux distribution
- [x] Google Chrome [Latest 2 versions]
- [x] Mozilla Firefox [Latest 2 versions]
- [x] Microsoft Edge [Latest 2 versions]
- [ ] Apple Safari [Latest 2 versions]
- [ ] Microsoft Internet Explorer 11

### CODE REVIEW

File: packages/opc-input-chip/README.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  myInputChipRef.chips = [
+    { id: 1, name: 'Test chip', isRemovable: true }
+  ];
+```
+
+### For Angular
+- In your app.module include the `CUSTOM_ELEMENTS_SCHEMA` and import the component
+```js
+import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import '@one-platform/opc-timeline';
```
Comment (hybridx) [inline packages/opc-input-chip/README.md:39]: @gisumit 
```suggestion
import '@one-platform/opc-input-chip';
```

File: packages/opc-input-chip/src/opc-input-chip.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  text-transform: opc-get-var(textTransform, lowercase);
+  background: opc-get-var(backgroundColor, $opc-color--blue-100);
+  color: opc-get-var(color, $opc-color--blue-400);
+  font-weight: opc-get-var(fontWeight, 400);
+  border-radius: 50px;
+  padding: 4px 12px 4px 12px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
```
Comment (hybridx) [inline packages/opc-input-chip/src/opc-input-chip.scss:33]: Can this be combined like so?

```suggestion
.input-chip {
  border: 1px solid;
  border-color: opc-get-var(borderColor, $opc-color--black-300);
  padding: 4px 1rem;
  min-height: 50px;
  border-radius: 5px;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  .chip {
  height: 28px;
  margin: 4px;
  cursor: pointer;
  text-transform: opc-get-var(textTransform, lowercase);
  background: opc-get-var(backgroundColor, $opc-color--blue-100);
  color: opc-get-var(color, $opc-color--blue-400);
  font-weight: opc-get-var(fontWeight, 400);
  border-radius: 50px;
  padding: 4px 12px 4px 12px;
  display: flex;
  justify-content: center;
  align-items: center;
}
}

```

File: packages/opc-input-chip/src/opc-input-chip.scss [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+input {
+  flex: 1;
+  margin: 4px;
+  border: none;
+  outline: none;
+  font-size: 1rem;
+  background-color: transparent;
+  color: opc-get-var(textColor, $opc-color--black-900);
+}
```
Comment (hybridx) [inline packages/opc-input-chip/src/opc-input-chip.scss:43]: Same as a above

**Review by hybridx — CHANGES_REQUESTED** [review]

File: packages/opc-input-chip/src/opc-input-chip.scss (hunk 3 above)
Comment (gisumit) [reply packages/opc-input-chip/src/opc-input-chip.scss]: yeah updated

File: packages/opc-input-chip/README.md (hunk 1 above)
Comment (gisumit) [reply packages/opc-input-chip/README.md]: updated and resolving.

File: packages/opc-input-chip/src/opc-input-chip.scss (hunk 2 above)
Comment (gisumit) [reply packages/opc-input-chip/src/opc-input-chip.scss]: updated and resolving.

File: packages/opc-input-chip/src/opc-input-chip.scss [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  input {
+    flex: 1;
+    margin: 4px;
+    border: none;
+    outline: none;
+    font-size: 1rem;
+    background-color: transparent;
+    color: opc-get-var(textColor, $opc-color--black-900);
+  }
```
Comment (hybridx) [inline packages/opc-input-chip/src/opc-input-chip.scss:49]: I guess this can also be moved inside

**Review by hybridx — CHANGES_REQUESTED** [review]

**Review by hybridx — APPROVED** [review]

File: packages/opc-input-chip/src/opc-input-chip.scss (hunk 4 above)
Comment (gisumit) [reply packages/opc-input-chip/src/opc-input-chip.scss]: nested inside of ```opc-input-chip```  container

File: packages/opc-input-chip/README.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+```
+
+## Themes
+- Only default themes
+
+## ### CSS Variables
+
+| CSS Variables   | Value (default) |
+|---------|----------------------|
+|--opc-inputChip--borderColor:| $opc-color--black-300, #D2D2D2|
```
Comment (diwanshi) [inline packages/opc-input-chip/README.md:107]: use format like this --opc-input-chip--BorderColor

File: packages/opc-input-chip/src/opc-input-chip.scss [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  padding: 4px 1rem;
+  min-height: 50px;
+  border-radius: 5px;
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+
+  .opc-input {
+    flex: 1;
+    margin: 4px;
```
Comment (diwanshi) [inline packages/opc-input-chip/src/opc-input-chip.scss:21]: can u use global css variable here?

Comment (diwanshi) [reply packages/opc-input-chip/src/opc-input-chip.scss:21]: u can use calc function to make it 4px

**Review by diwanshi — CHANGES_REQUESTED** [review]

File: packages/opc-input-chip/README.md (hunk 5 above)
Comment (gisumit) [reply packages/opc-input-chip/README.md]: converted to --opc-input-chip--BorderColor

File: packages/opc-input-chip/src/opc-input-chip.scss (hunk 6 above)
Comment (gisumit) [reply packages/opc-input-chip/src/opc-input-chip.scss]: yeah created a map to hold reusable units. I tried using the following interpretation but it wasn't working

```scss
$input-chip-units: (
  margin: 4,
  border-width: 1,
);

.opc-input {
  // to string interpolation with unit as px/rem etc
 // this doesnt work 
  margin: #{map-get($input-chip-units, margin)px}
}

```
So Created a map with units

```scss
$input-chip-units: (
  margin: 4px,
  border-width: 1px,
);

```

**Comment by gisumit** [comment]
@diwanshi Can you review the changes

**Review by diwanshi — APPROVED** [review]

**Pull request merged**